apps/api/src/app/integrations/usecases/create-integration/create-integration.usecase.ts

```diff
@@ -26,7 +28,20 @@ export class CreateIntegration {
   ) {}
 
   async execute(command: CreateIntegrationCommand): Promise<IntegrationEntity> {
-    let response: IntegrationEntity;
+    if (command.identifier) {
+      const existingIntegrationWithIdentifier = await this.integrationRepository.findOne({
+        _organizationId: command.organizationId,
+        identifier: command.identifier,
+      });
+
+      if (existingIntegrationWithIdentifier) {
+        throw new BadRequestException('Integration with identifier already exists');
+      }
+    }
+
+    if (command.active && Object.keys(command.credentials ?? {}).length === 0) {
+      throw new BadRequestException('The credentials are required to activate the integration');
+    }
```

Comment:
If someone tries to activate the integration without the credentials throw an error. I know this check is silly but I don't know the better way and also we have the email test check below implemented too.

Reply:
This is enough for now. Should be on the validation layer but there the complication is bigger to make it work properly.

Reply:
No, I mean in the `CreateIntegrationRequestDto` when we validate using `class-validator`.

Reply:
ok, got it, not sure where is the best place for that more complex logic, but I would say probably in the use-case.